index.js

```diff
@@ -85,12 +95,45 @@
     return result ? result[1] : "";
   }
 
+  function getComponentsList(imports) {
+    const dxComponents = Object.keys(imports).filter((cmp) => cmp.startsWith('Dx'));
+    const appComponents = Object.keys(imports).filter((key) => imports[key].source.endsWith('.vue'));
+    return [...dxComponents, ...appComponents];
+  }
+
+  async function getCompositionApiSFC(source, template) {
+    const compiledScript = compileScript(parse(source).descriptor, {id: 'demo-'});
+    const compiledTemplate = compileTemplate({source: template ,id: 'demo-'});
+    const templateImports = compiledTemplate.code.replace(/export function.*/s, '');
+    const templateRenderFn = compiledTemplate.code.replace(/^.*export function\s*/s, '');
+
+    const componentsList = getComponentsList(compiledScript.imports);
+
+    const compiledScriptContent = templateImports.trim()
+        + '\n'
+        + compiledScript.content
+            .replace(
+                /defineComponent\(\{/,
+                `defineComponent({\n
+        components: {
+              ${componentsList.join(',\n').trim()}
+         },
+        ${templateRenderFn},
+    `).replace(/return __returned__/, 'return {...__returned__};');
+
+    return ts.transpileModule(
```

Comment:
We also should access the ts.transpileModule inside our plugin. In the WG env we do it outside using the "compiledScriptContent" (non-transpilled) result.